### PR TITLE
[EICNET-984]feature: Add into search the possibility to filter by interests and group/contents

### PR DESCRIPTION
**Pull request by Dromadoros**
New fields in block configuration to be able to filter on SOLR via search API + integrate it into react

how to test it:

- [x] Go to your eic theme and npm install && npm run react-watch
- [x] Go to block layout, add EIC Search Overview to content region
- [x] Configuration : you need to enable "Add filter for my groups/contents" and "Add filter for my topics of interests"
- [x] Clear your cache and go to the homepage
- [x] You can now use your contents and should filter your content that your are the author + your groups. Or for interests, topics that match with yours

### CODE REVIEW

File: lib/modules/eic_search/src/Controller/SolrSearchController.php [hunk 1]
```diff
@@ -139,4 +154,65 @@ private function getFacetsQuery($facets_value): string {
     return $facets_query;
   }
 
+  /**
+   * @param $fq
+   * @param $interests
```
Comment (4hardfire) [inline lib/modules/eic_search/src/Controller/SolrSearchController.php:159]: grumphp should complain about this.

File: lib/modules/eic_search/src/Controller/SolrSearchController.php [hunk 2]
```diff
@@ -139,4 +154,65 @@ private function getFacetsQuery($facets_value): string {
...
+    }
+
+    $user_topics_string = implode(' OR ', $user_topics_id);
+
+    $fq .= " AND (itm_group_field_vocab_topics:($user_topics_string) OR itm_content_field_vocab_topics:($user_topics_string))";
+  }
+
+  /**
+   * @param $fq
+   * @param $interests
```
Comment (4hardfire) [inline lib/modules/eic_search/src/Controller/SolrSearchController.php:189]: grumphp should complain about this.

**Review by 4hardfire — CHANGES_REQUESTED** [review]

File: lib/modules/eic_search/src/Controller/SolrSearchController.php (hunk 2 above)
Comment (Dromadoros) [reply lib/modules/eic_search/src/Controller/SolrSearchController.php:189]: You're right my friend

File: lib/modules/eic_search/src/Controller/SolrSearchController.php (hunk 1 above)
Comment (Dromadoros) [reply lib/modules/eic_search/src/Controller/SolrSearchController.php]: Done 👍

File: lib/modules/eic_search/src/Controller/SolrSearchController.php (hunk 2 above)
Comment (Dromadoros) [reply lib/modules/eic_search/src/Controller/SolrSearchController.php]: Done 👍

**Review by 4hardfire — APPROVED** [review]
Looks good to me!

**Pull request merged**